src/countdown-timer/countdown-timer.tsx

```diff
@@ -0,0 +1,172 @@
+            performOnNotifyHandler();
+        } else if (timer < notifyTiming) {
+            performOnDurationHandler();
+        }
+    }, [timer]);
+
+    useEffect(() => {
+        const y = getOffsetY();
+        setOffsetY(y);
+    }, [isMobile, wrapperRef.current]);
+
+    // =============================================================================
+    // EVENT HANDLERS
+    // =============================================================================
+    function handleScrollEvent() {
+        setScrolling(true);
+        handleEndScroll();
+    }
+
+    const handleEndScroll = useMemo(
+        () => debounce(() => setScrolling(false), 250),
+        []
+    );
+
+    // =============================================================================
+    // HELPER FUNCTIONS
+    // =============================================================================
+    const performOnDurationHandler = () => {
+        if (onDuration) {
+            const { minutes, seconds } = TimeHelper.convertSecondsToTime(timer);
+
+            onDuration({ minutes, seconds });
+        }
+    };
+
+    const performOnNotifyHandler = () => {
+        if (onNotify) onNotify();
+    };
+
+    const performOnFinishHandler = () => {
+        setIsPlaying(false);
+
+        if (onFinish) onFinish();
+    };
+
+    function getOffsetY() {
+        const DEFAULT_DESKTOP_TABLET_POSITION = 168;
+        const DEFAULT_MOBILE_POSITION = 80;
+
+        const offsetY = offset?.top
+            ? offset.top
+            : isMobile
+            ? DEFAULT_MOBILE_POSITION
+            : DEFAULT_DESKTOP_TABLET_POSITION;
+
+        return offsetY;
+    }
+
+    // =============================================================================
+    // RENDER FUNCTION
+    // =============================================================================
+
+    const renderCountdown = (position: PositionStyle) => {
+        const { minutes, seconds } = TimeHelper.convertSecondsToTime(timer);
+        const clientRect = wrapperRef.current?.getBoundingClientRect();
+
+        return (
+            <Countdown
+                data-testid={testId}
+                data-id={
+                    position === "relative"
+                        ? `countdown-wrapper`
+                        : `fixed-countdown-wrapper`
+                }
+                ref={position === "relative" ? wrapperRef : undefined}
+                inert={onScreen ? undefined : ""}
+                $position={position}
+                $pinned={position === "fixed" && !onScreen}
+                $opacity={position === "relative" && !onScreen}
+                $scroll={scrolling}
+                $warn={timer <= notifyTiming}
+                $top={offsetY}
+                $left={offset?.left ?? clientRect?.x}
+            >
+                <ClockIcon />
+                <TimeLeft>Time left:</TimeLeft>
+                <Time>
+                    {minutes} mins {String(seconds).padStart(2, "0")} secs
```

Comment:
to handle the singular/plural form here for minute and second

example:
0 mins
1 min
2 mins